### ISSUE TITLE
Its leaking too much information

### ISSUE THREAD

**Issue by dharmendrajadon** [lucadegasperi/oauth2-server-laravel#410]
I'm getting some more than 2Mb information from Authorizer::getAuthCodeRequestParams()
It is even getting other controller methods and information
Route::api(array('version' => 'v1'), function () {
	Route::resource('dosomething', 'SomeController'));
});
/////////////////////////////////////////
Route::get('oauth/authorize', ['before' => 'check-authorization-params|auth', function() {
    // display a form where the user can authorize the client to access it's data
    Log::info(Authorizer::getAuthCodeRequestParams());// Log this information and see yourself
    return View::make('oauth/authorization-form', array('params' => Authorizer::getAuthCodeRequestParams()));
}]);

**Comment by lucadegasperi** [comment]
There's probably an error that doesn't get caught.

**Comment by dharmendrajadon** [comment]
When I tried to Log::info(Authorizer::getAuthCodeRequestParams()), it says memory exhausted(even 256 Mb is less). Then I increased memory to process and Got some more than 2Mb data being holded in Authorizer::getAuthCodeRequestParams() with lots of other controller classes, other methods, even User Controller for authenticating user, each and every controller and routing information included.

**Comment by lucadegasperi** [comment]
Weird. I don't have any other user reporting this, It might definitively due to the server configuration.

**Comment by dharmendrajadon** [comment]
Maybe, they haven't tried it so far! OK! I used Postman to test API. Everything appeared fine till I'm asked to approve the request(Approve or deny). On clicking Approve, it throws an error client_id is missing. I traced it and found that In second attempt on form submission(After clicking Approve or Deny), I gets parameter missing error. From here, I got an idea to log all process flow. When we use "Authorizer::getAuthCodeRequestParams()" to extract information on Approve/Deny page it lacks in client_id parameter. It do contain client(as Object client - not as client_id), request_uri, request_type, state. But, this "Authorizer::getAuthCodeRequestParams()" holds too much leaked information too.
